File: experiments_dataset_vul_detect_cwe_classification/dataset/vul_dataset/python/non-vul/code_before24.py
# Source: Row 539 in ./dataset/CVEfixes/Analysis/results/Python/df_python_all.xlsx

def setUp(self):
    self.reactor = ThreadedMemoryReactorClock()
    self.hs_clock = Clock(self.reactor)
    self.homeserver = setup_test_homeserver(
        self.addCleanup,
        federation_http_client=None,
        clock=self.hs_clock,
        reactor=self.reactor,
    )

File: experiments_dataset_vul_detect_cwe_classification/dataset/vul_dataset/cpp/non-vul/code_before24.cpp
// Source: Row 1878 in ./dataset/CVEfixes/Analysis/results/C++/df_cpp_all.xlsx

FdOutStream::FdOutStream(int fd_, bool blocking_, int timeoutms_, size_t bufSize_)
  : fd(fd_), blocking(blocking_), timeoutms(timeoutms_),
    bufSize(bufSize_ ? bufSize_ : DEFAULT_BUF_SIZE), offset(0)
{
  ptr = start = sentUpTo = new U8[bufSize];
  end = start + bufSize;

  gettimeofday(&lastWrite, NULL);
}

